enterprise/signals.py

```diff
@@ -349,16 +351,23 @@ def enterprise_unenrollment_receiver(sender, **kwargs):     # pylint: disable=un
     """
     enrollment = kwargs.get('enrollment')
     enterprise_enrollment = models.EnterpriseCourseEnrollment.objects.filter(
-        course_id=enrollment.course.course_key,
-        enterprise_customer_user__user_id=enrollment.user.id,
+        course_id=enrollment.course_id,
+        enterprise_customer_user__user_id=enrollment.enterprise_customer_user.user_id,
     ).first()
     if enterprise_enrollment:
         logger.info(
-            f"Marking EnterpriseCourseEnrollment as unenrolled for user {enrollment.user} and "
-            f"course {enrollment.course.course_key}"
+            f"Marking EnterpriseCourseEnrollment as unenrolled for user "
+            f"{enrollment.enterprise_customer_user.user_id} and course {enrollment.course_id}"
         )
         enterprise_enrollment.unenrolled_at = localized_utcnow()
         enterprise_enrollment.save()
+        get_producer().send(
+            signal=COURSE_UNENROLLMENT_COMPLETED,
```

Comment:
this is a generic unenrollment signal but you only fire it for enterprise enrollments - i'd probably produce an event for all of these signals, not just the enterprise ones. i think this event is emitted somewhere upstream and that is where we should put that signal on the bus